.github/workflows/ci_cd.yml

```diff
@@ -198,26 +198,28 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@v3
 
-      # ℹ️Step only applies for frontend and catalog images.
-      # Frontend downloads translations and catalog runs `just` recipes.
+      # Sets up `just` and Node.js.
+      # Node.js is needed for the frontend to download translations.
+      # `just` is needed to run the recipes and set build args.
       - name: Setup CI env
-        if: matrix.image == 'frontend' || matrix.image == 'catalog'
         uses: ./.github/actions/setup-env
         with:
           setup_python: false
-          # Python is not needed to build the image.
-          install_recipe: node-install
+          setup_nodejs: ${{ matrix.image == 'frontend' && 'true' || 'false' }}
+          install_recipe: ${{ matrix.image == 'frontend' && 'node-install' || '' }}
 
-      # ℹ️While this step only applies for catalog image, it is run for all.
-      # This step
-      # - sets the Python version used by the catalog image
-      # - sets the Airflow version used by the catalog image
-      - name: Prepare catalog for building
-        id: prepare-catalog
-        # if: matrix.image == 'catalog'
+      # Sets build args specifying versions needed to build Docker image.
+      - name: Prepare build args
+        id: prepare-build-args
         run: |
-          echo "python_version=$(just catalog/py-version)" >> "$GITHUB_OUTPUT"
-          echo "airflow_version=$(just catalog/airflow-version)" >> "$GITHUB_OUTPUT"
+          cat <<EOF >> "$GITHUB_OUTPUT"
+          catalog_py_version=$(just catalog/py-version)
+          catalog_airflow_version=$(just catalog/airflow-version)
+          api_py_version=$(just api/py-version)
+          ing_py_version=$(just ingestion_server/py-version)
+          frontend_node_version=$(just frontend/node-version)
+          frontend_pnpm_version=$(just frontend/pnpm-version)
+          EOF
```

Comment:
This might be overdoing it, but would it make sense to have a single just recipe at the top level which prince this information? Maybe `just versions`? Then this step could be `just versions >> "$GITHUB_OUTPUT"`

Reply:
Updated in 4b9dc14.